src/store/modules/product/actions.ts

```diff
@@ -46,6 +46,39 @@ const actions: ActionTree<ProductState, RootState> = {
     // TODO Handle specific error
     return resp;
   },
+  async fetchProductDetails({ commit, state }, { productIds }) {
+    const cachedProductIds = Object.keys(state.cached);
+    const productIdFilter = productIds.reduce((filter: string, productId: any) => {
+      if(cachedProductIds.includes(productId)) {
+        return filter;
+      }
+      else {
+        if(filter !== '') filter += ' OR '
+        return filter += productId;
+      }
+    }, '');
+    if (productIdFilter === '') return;
+    const resp = await ProductService.fetchProducts({
+      "filters": ['productId: (' + productIdFilter + ')']
```

Comment:
We should also pass the viewSize here that should be equal to the number of productIds passed because the API default returns only 10 items.

Reply:
Improved Sir.